README.md

```diff
@@ -1,6 +1,168 @@
+
+## How does it work?
+
+Strimzi Drain Cleaner using Kubernetes Admission Control features and Validating Web-hooks to find out when something tries to evict the Kafka or Zookeeper pods are.
+It annotates them with the `strimzi.io/manual-rolling-update` annotation which will tell Strimzi Cluster Operator that this pod needs to be restarted.
+Strimzi will roll it in the next reconciliation using its algorithms which make sure the cluster is available.
+**This is supported from Strimzi 0.21.0.**
+
+## Deployment
+
+If you want to use this only to Kafka and not to ZooKeeper, you can edit the Deployment and remove the `--zookeeper` option.
+
+### On OpenShift
+
+On OpenShift, you can have the certificates needed for the web-hook generated automatically and injected into the pod / web-hook configuration.
+To install the Drain Cleaner on OpenShift, use the `./deploy/openshift` directory:
+
+```
+kubectl apply -f ./deploy/openshift
+```
+
+### On Kubernetes with CertManager
+
+On Kubernetes, when you use Cert Manager, you can have the certificates needed for the web-hook generated automatically and injected into the pod / web-hook configuration.
+To install the Drain Cleaner on Kubernetes with installed CertManager, use the `./deploy/certmanager` directory:
+
+```
+kubectl apply -f ./deploy/certmanager
+```
+
+### On Kubernetes without CertManager
+
+On Kubernetes, when you do not use Cert Manager, the certificates needed for the web-hook need to be generated manually.
+Follow the instructions in `./deploy/kubernetes` directory.
+
+## See it in action
+
+You can easily test how it works:
+* Install Strimzi or AMQ Streams in your cluster
+* Deploy Kafka cluster with Pod Disruption Budget configuration having `masUnavailable` set to `0` as shown in the example above
+* Install the Drain Cleaner
+* Drain one of the Kubernetes nodes with one of the Kafka or ZooKeeper pods
+    ```
+    kubectl drain <worker-node> --delete-emptydir-data --ignore-daemonsets --timeout=6000s --force
+    ```
+* Watch how it works:
+    * The `kubetl drain` command will wait for the Kafka / ZooKeeper to be drained
+    * The Drain Cleaner log should show how it gets the eviction events
+    * Strimzi Cluster Operator log should show how it rolls the pods which are being evicted
+    
+## Build 
+
+This project uses [Quarkus, the Supersonic Subatomic Java Framework](https://quarkus.io/).
+
+### Running the application in dev mode
+
+You can run your application in dev mode that enables live coding using:
+```shell script
+./mvnw compile quarkus:dev
+```
+
+### Creating a native executable
+
+You can create a native executable using: 
+```shell script
+./mvnw package -Pnative
+```
+
+Or you can run the native executable build for Linux in a container using: 
+```shell script
+./mvnw package -Pnative -Dquarkus.native.container-build=true
+```
+
+You can then execute your native executable with: `./target/strimzi-cistic-odpadu-1.0.0-SNAPSHOT-runner`.
```

Comment:
wondering if the artifact should be `strimzi-drain-cleaner`

Reply:
This contributes the original code. We should change it in the future.